internal/helper/helper.go

```diff
@@ -284,12 +277,47 @@ func SafeQuantitySubtraction(orig, subtract quantity.Size) quantity.Size {
 	return orig - subtract
 }
 
+// CreateTarArchive places all of the files from a source directory into a tar.
+// Currently supported are uncompressed tar archives and the following
+// compression types: zip, gzip, xz bzip2, zstd
+func CreateTarArchive(src, dest, compression string, verbose, debug bool) error {
+	tarCommand := *exec.Command("tar",
+		"--directory",
+		src,
+		"--xattrs",
+		"--create",
+		"--file",
+		dest,
+		".",
+	)
+	fmt.Println(tarCommand.String())
+	if debug {
+		tarCommand.Args = append(tarCommand.Args, "--verbose")
+	}
+	tarOutput := SetCommandOutput(&tarCommand, debug)
+	if err := tarCommand.Run(); err != nil {
+		return fmt.Errorf("Error running \"tar\" command \"%s\". "+
+			"Error is \"%s\". Full output below:\n%s",
+			tarCommand.String(), err.Error(), tarOutput.String())
+	}
+	return nil
+}
+
 // ExtractTarArchive extracts all the files from a tar. Currently supported are
 // uncompressed tar archives and the following compression types: zip, gzip, xz
 // bzip2, zstd
 func ExtractTarArchive(src, dest string, verbose, debug bool) error {
-	// magic numbers are used to determine the compression type of
-	// the tar archive
+	tarCommand := *exec.Command("tar",
+		"--extract",
+		"--file",
+		src,
+		"--directory",
+		dest,
+	)
```

Comment:
This invocation needs xattrs too I'm pretty sure.

Reply:
Fixed.